rust/src/nfs/nfs2_records.rs

```diff
@@ -96,3 +96,159 @@ pub fn parse_nfs2_attribs(i: &[u8]) -> IResult<&[u8], Nfs2Attributes> {
     let attrs = Nfs2Attributes { atype, asize };
     Ok((i, attrs))
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::nfs::nfs2_records::*;
+
+    #[test]
+    fn test_nfs2_handle() {
+
+        // file_handle bytes
+        let buf: &[u8] = &[
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5a,
+            0x00, 0x00, 0x00, 0x29, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29
+        ];
+
+        let result = parse_nfs2_handle(buf).unwrap();
+        match result {
+            (r, res) => {
+                assert_eq!(r.len(), 0);
+                assert_eq!(res.value, buf);
+            }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_request_lookup() {
+
+        let buf: &[u8] = &[
+        // [file_handle]
+            0x00, 0x10, 0x10, 0x85, 0x00, 0x00, 0x03, 0xe7,
+            0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, 0xb2, 0x5a,
+            0x00, 0x00, 0x00, 0x29, 0x00, 0x0a, 0x00, 0x00,
+            0x00, 0x00, 0xb2, 0x5a, 0x00, 0x00, 0x00, 0x29,
+        // [name]
+            0x00, 0x00, 0x00, 0x02, 0x61, 0x6d, 0x00, 0x00,
```

Comment:
IMO here we have 3 fields in one line, it's not clear what's the length of `name_contents` since we assert against it, it would take time to grab the relevant bytes and check for the value since we have a _not clear_ hidden `_name_padding` byte.

Reply:
`rustfmt` suggested this style in my code: https://github.com/OISF/suricata/blob/5dec901bb7ae6a37f873c42260653f20279222cd/rust/src/pgsql/parser.rs#L1012

Although I can't say that I love it, it does allow for having portions of data identified by comments without adding too many additional lines to the tests...

Reply:
that what `rustfmt` suggested with my code, and I cannot say that I love it too :) , as I mentioned I dont know how to count the trade off with regard to this.

Reply:
Jason has recommended/ asked me to run rustfmt on my pgsql code, so I think we're ok with the outcome, even if sometimes it wouldn't be our personal choice :P